### PR TITLE
add map reduce test

### PR DESCRIPTION
**Pull request by krish8484**

### CODE REVIEW

File: Protos/api.proto [hunk 1]
```diff
@@ -87,5 +87,10 @@ message TaskDataParam {
     oneof dataParam {
         Future future = 1;
         bytes data = 2;
+        FutureList futures = 3;
```
Comment (swarnim-vyas) [inline Protos/api.proto]: It needs to have both Future and FutureList?

Comment (krish8484) [reply Protos/api.proto]: one of them

Comment (krish8484) [reply Protos/api.proto]: Future is kept for backward compatibility

**Review by swarnim-vyas — APPROVED** [review]

File: Protos/api.proto (hunk 1 above)
Comment (aadhithNarayan) [inline Protos/api.proto]: Why is this needed and where it is used?

Comment (krish8484) [reply Protos/api.proto]: used in addition operation

**Review by aadhithNarayan — APPROVED** [review]
